tools/build_script_generator/compilation_db/resources/builder_compilation_db.py.in

```diff
@@ -0,0 +1,108 @@
+# Copyright (c) 2019, Niklas Hauser
+#
+# This file is part of the modm project.
+#
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import os, re, json
+from collections import defaultdict
+from pathlib import Path
+
+env = defaultdict(list)
+
+env["CONFIG_PROJECT_NAME"] = "{{ options[":build:project.name"] }}"
+env["CONFIG_BUILD_BASE"] = "{{ build_path }}"
+env["CONFIG_BUILD_PROFILE"] = "release"
+
+env["BUILDPATH"] = os.path.join(env["CONFIG_BUILD_BASE"], env["CONFIG_BUILD_PROFILE"])
+env["BASEPATH"] = os.path.abspath(".")
+
+generated_paths = {{ generated_paths }}
+
+for path in generated_paths:
+    with open("{}/compilation_db.py".format(path)) as compfile:
+        exec(compfile.read())
+
+source_flag_map = {{common_source_flag_map}}
+env["CPPDEFINES"] = map(lambda d: "-D"+d, env["CPPDEFINES"])
+
+# FIXME: Search for application files
+project_files = ["main.cpp"]
```

Comment:
Will this suffice it?
```
project_files = Path(".").glob("*.cpp")
```

Reply:
Answering myself:
No. We need something like this:
```
project_files =  list(map(str, map(Path.resolve, Path(".").glob("*.cpp"))))
```

Reply:
Yeah, but recursive and ignoring the generated folders ;-P
There's [a SCons tool](https://github.com/modm-io/modm/blob/develop/tools/build_script_generator/scons/site_tools/find_files.py#L20) for it, probably just going to copy that…